test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnVerbs.java

```diff
@@ -330,6 +353,105 @@ public static HapiContractUpdate contractUpdate(String contract) {
 		return new HapiContractUpdate(contract);
 	}
 
+	/**
+	 * This method enables the developer to upload one or many contract(s) bytecode(s), but without the ability to chain methods
+	 * related to HapiFileCreate.
+	 */
+	public static HapiSpecOperation uploadInitCode(final String... contractsNames) {
+		return withOpContext((spec, ctxLog) -> {
+			List<HapiSpecOperation> ops = new ArrayList<>();
+			for (String contractName : contractsNames) {
+				final var path = getResourcePath(contractName, ".bin");
+				final var file = new HapiFileCreate(contractName);
+				final var updatedFile = updateLargeFile(GENESIS, contractName, extractByteCode(path));
+				ops.add(file);
+				ops.add(updatedFile);
+			}
+			allRunFor(spec, ops);
+		});
+	}
+
+	/**This method enables the developer to upload contract bytecode with the ability to chain methods, related to HapiFileCreate.
+	 The execution of the method requires the op context, therefore the invocation is not straightforward.
+	 Example:
+	 withOpContext(
+	 (spec, noOp) -> {
+	 chainedFileCreate(contract, spec).hasPrecheck(TRANSACTION_OVERSIZE);
+	 chainedFileCreate(contract, spec).contents("").key(KEY_LIST);
+	 }
+	 )
+	 */
+	public static HapiFileCreate uploadInitCode(final String contractName, final HapiApiSpec spec) {
+		final var path = getResourcePath(contractName, ".bin");
+		final var file = new HapiFileCreate(contractName);
+		sourcing(() -> file).execFor(spec);
+		sourcing(() -> updateLargeFile(GENESIS, contractName, extractByteCode(path))).execFor(spec);
+		return file;
+	}
+
+	/*  Note to the reviewer:
+		This method is temporarily named with the "new" prefix, as soon as the implementation is approved and the legacy
+		contractCreate() is entirely replaced, this method will be renamed to "contractCreate"
+	 */
+	public static HapiContractCreate newContractCreate(final String contractName, final Object... constructorParams) {
+		if (constructorParams.length > 0) {
+			final var constructorABI = getABIFor(CONSTRUCTOR, EMPTY, contractName);
+			return new HapiContractCreate(contractName, constructorABI, constructorParams).bytecode(contractName);
+		} else {
+			return new HapiContractCreate(contractName).bytecode(contractName);
+		}
+	}
```

Comment:
Since this is the final version of the PR we should remove the notes to the reviewer (at least from the code base, if we need to keep notes, we can add them as PR description). Instead we can document what params the method accepts and how it behaves, what the return type is. We can use javadoc comments style for that.
I think we can safely rename the `newContractCreate` method to `contractCreate`. There is no other instance with the same signature so there is no problem in having more than one method with the same name. If we need to keep the legacy `contractCreate()` methods that's fine.

Reply:
Solved